demo/lightning/image-segmentation/pytorch_loader.py

```diff
@@ -0,0 +1,203 @@
+"""
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+
+import io
+import random
+import dataflux_core
+import numpy as np
+import scipy.ndimage
+from google.api_core.client_info import ClientInfo
+from google.cloud import storage
+from torch.utils.data import Dataset
+
+from dataflux_pytorch import dataflux_mapstyle_dataset
+
+
+class DatafluxPytTrain(Dataset):
+
+    def __init__(
+        self,
+        project_name,
+        bucket_name,
+        config=dataflux_mapstyle_dataset.Config(),
+        storage_client=None,
+        **kwargs,
+    ):
+
+        self.train_transforms = kwargs["transforms"]
+        patch_size, oversampling = kwargs["patch_size"], kwargs["oversampling"]
+        images_prefix, labels_prefix = kwargs["images_prefix"], kwargs[
+            "labels_prefix"]
+        self.patch_size = patch_size
+        self.rand_crop = RandBalancedCrop(patch_size=patch_size,
+                                          oversampling=oversampling)
+
+        # Dataflux-specific setup.
+        self.project_name = project_name
+        self.bucket_name = bucket_name
+        self.config = config
+        self.dataflux_download_optimization_params = (
+            dataflux_core.download.DataFluxDownloadOptimizationParams(
+                max_composite_object_size=self.config.max_composite_object_size
+            ))
+        if not storage_client:
+            self.storage_client = storage.Client(
+                project=project_name,
+                client_info=ClientInfo(user_agent="dataflux/0.0"),
```

Comment:
client_info needs not to be manually set now, but just a client passed in. https://github.com/GoogleCloudPlatform/dataflux-client-python/blob/main/dataflux_core/fast_list.py#L176

Reply:
Removed here and in `demo/image-segmentation/pytorch_loader.py`